src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +269,51 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.Now;
+            _timer = new Timer( async o => await SendHeartBeat(o).ConfigureAwait(false),
+                null,
+                (int)(StreamWithTimeout.DefaultWriteTimeout.TotalMilliseconds / 3),
+                Timeout.InfiniteTimeSpan.Milliseconds);
+        }
+
+        internal async Task SendHeartBeat(object _)
```

Comment:
I think you need to create `HandleHeartbeat` method that matches TimerCallback signature and probably use AsyncHelpers there or make the code sync here (but this will be more complex and I do not think is needed).

Reply:
done